packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -470,6 +482,10 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		];
 
 		if (maxCount > 0) {
+			// This logic is not perfect, as there can be executions after the
+			// maxCount-th execution that can't be pruned because they are not
+			// in an end state. Therefore we can end up with more than maxCount
+			// executions. But it's good enough for now.
```

Comment:
Ever since pruning exists, it has never looked at execution status. So instead of commenting here, we might want to update [the docs](https://docs.n8n.io/hosting/scaling/execution-data/#enable-data-pruning) to the effect that only finished executions are pruned, and as a side note mention that is why the result of pruning based on max count might not be exact. Same for the pruning-related env var descriptions in `schema.ts`

Reply:
Created docs [PR](https://github.com/n8n-io/n8n-docs/pull/1633)